Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -359,5 +511,235 @@ bool ComponentInfo::isStructuredBank(const size_t componentIndex) const {
   return !isDetector(componentIndex) && (*m_isStructuredBank)[rangesIndex];
 }
 
+/**
+ * Get the number of scans for component index
+ * @param index : Component Index
+ * @return Number of scans for component index
+ */
+size_t ComponentInfo::scanCount(const size_t index) const {
+  if (m_detectorInfo && isDetector(index))
+    return m_detectorInfo->scanCount(index);
+  if (!m_scanCounts) {
+    return 1;
+  }
+  if (m_isSyncScan) {
+    return (*m_scanCounts)[0];
+  }
+  return (*m_scanCounts)[compOffsetIndex(index)];
+}
+
+size_t ComponentInfo::scanSize() const {
+  const auto detectorScanSize = m_detectorInfo ? m_detectorInfo->scanSize() : 0;
+  if (!m_positions)
+    return 0 + detectorScanSize;
+  return m_positions->size() + detectorScanSize;
+}
+
+bool ComponentInfo::isScanning() const {
+  if (m_detectorInfo && m_detectorInfo->isScanning())
+    return true;
+  else if (!m_positions || !m_componentRanges)
+    return false;
+  else
+    // TODO what if you have a scan that causes no position change, such as a
+    // rotation. Not scanning???
```

Comment:
Any scan will (should?) duplicate all parameters, even if they are kept fixed. At least that is what `DetectorInfo` is doing, I think.

Reply:
If that's happening, that is OK. I think you're right, this should happen in the `merge`